src/deployLambda.js

```diff
@@ -43,11 +60,15 @@ const updateOrCreateLambda = async (aws, params = {}) => {
         Runtime: params.runtime || 'nodejs12.x',
         Environment: {
           Variables: params.env || {}
-        }
+        },
+        VpcConfig: vpcConfig
       }
 
       await lambda.updateFunctionConfiguration(updateFunctionConfigurationParams).promise()
 
+      // Updates (like vpc changes) need to complete before calling updateFunctionCode
+      await lambda.waitFor('functionUpdated', { FunctionName: params.lambdaName }).promise()
```

Comment:
@hypexr That's a good one! do you know where I can find a list of events I can "wait for" in the aws sdk? 🤔

Reply:
@eahefnawy There are only a few lambda specific ones

functionExists
functionActive
functionUpdated

https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/Lambda.html#waitFor-property 

I'm not sure if there's anything more universal than that for the whole aws sdk.  It's not intuitive that a wait would be necessary before calling another config/code change, but it makes sense and it consistently failed when adding the VPC configuration to a lambda that was previously deployed without it. Interestingly updating a lambda with a vpc config takes some time (5+ seconds and in the console as well), but if it's created initially with the VPC the time difference is minimal.